tests/vision/checks/train_test_validation/image_dataset_drift_test.py

```diff
@@ -72,137 +72,84 @@ def test_no_drift_grayscale_cramer(mnist_dataset_train, device):
 
     # Act
-    result = check.run(train, test, random_state=42, device=device)
+    result = check.run(coco_visiondata_train, coco_visiondata_test)
     # Assert
     assert_that(result.value, has_entries({
-        'domain_classifier_auc': close_to(0.623, 0.001),
-        'domain_classifier_drift_score': close_to(0.247, 0.001),
+        'domain_classifier_auc': close_to(0.265, 0.001),
+        'domain_classifier_drift_score': close_to(0, 0.001),
         'domain_classifier_feature_importance': has_entries({
             'RMS Contrast': equal_to(0),
             'Brightness': close_to(0, 0.01),
             'Aspect Ratio': close_to(0, 0.01),
             'Area': close_to(0, 0.001),
             'Mean Red Relative Intensity': equal_to(0),
             'Mean Blue Relative Intensity': close_to(0, 0.001),
-            'Mean Green Relative Intensity': close_to(1, 0.001),
+            'Mean Green Relative Intensity': close_to(0, 0.001),
         })
     }))
+    assert_that(result.conditions_results[0], equal_condition_result(
+        is_pass=True,
+        name=f'Drift score is less than 0.3',
+        details=f'Drift score 0 is less than 0.3',
+    ))
 
 
-def test_with_drift_rgb(coco_train_dataloader, coco_test_dataloader, device):
+def test_with_drift_rgb(coco_visiondata_train, coco_visiondata_test):
     # Arrange
-    class DriftCoco(COCOData):
-        def batch_to_images(self, batch):
-            return pil_drift_formatter(batch)
+    def collate_generate_drift(batch):
+        images, labels = collate_without_model(batch)
+        return {'images': pil_drift_formatter(images), 'labels': labels}
 
-    train = DriftCoco(coco_train_dataloader)
-    test = COCOData(coco_test_dataloader)
+    drifted_train = replace_collate_fn_visiondata(coco_visiondata_train, collate_generate_drift)
 
-    check = ImageDatasetDrift(categorical_drift_method='PSI')
+    check = ImageDatasetDrift(categorical_drift_method='PSI').add_condition_drift_score_less_than()
     # Act
-    result = check.run(train, test, random_state=42, device=device)
+    result = check.run(drifted_train, coco_visiondata_test)
     # Assert
     assert_that(result.value, has_entries({
-        'domain_classifier_auc': close_to(0.908, 0.001),
-        'domain_classifier_drift_score': close_to(0.815, 0.001),
+        'domain_classifier_auc': close_to(0.972, 0.001),
+        'domain_classifier_drift_score': close_to(0.945, 0.001),
         'domain_classifier_feature_importance': has_entries({
-            'Brightness': close_to(1, 0.001),
-            'Aspect Ratio': equal_to(0),
-            'Area': equal_to(0),
-            'Mean Red Relative Intensity': close_to(0, 0.01),
-            'Mean Green Relative Intensity': close_to(0, 0.01),
-            'Mean Blue Relative Intensity': close_to(0, 0.01),
+            'Brightness': close_to(0.967, 0.001),
         })
     }))
-
-
-def test_condition_fail(coco_train_dataloader, coco_test_dataloader, device):
```

Comment:
Why not test conditions?

Reply:
merged it with above test to reduce run time